### DISCUSSION TITLE
How to fetch/calculate all account's staking and delegations?

### DISCUSSION THREAD

**Discussion by tktres** [aptos-labs/aptos-developer-discussions#114] [Questions] [answered]
I'd like to implement by code a function that gets an account address and returns all its stakes and delegations?
Ideally with identification of deposited funds and earned rewards

**Comment by kshitijc1506** [comment, marked as answer]
To manage and calculate an account's staking and delegations within the Aptos blockchain, you may need to implement custom logic. This involves searching through an account's transaction history to identify which delegation pools it has interacted with. For small-scale implementations, a brute force search could suffice. Here's a code snippet using the SDK:

`await aptos.getAccountTransactions({ accountAddress: "0x123" })`

For information regarding delegations, the [documentation](https://aptos.dev/nodes/validator-node/operator/delegation-pool-operations/#check-delegation-pool-information) provides insights into the various view functions accessible via the SDK. You could also use the [indexer](https://cloud.hasura.io/public/graphiql?endpoint=https://indexer.mainnet.aptoslabs.com/v1/graphql) directly too. 

Here's a potential approach:

1. Checking for Delegation Pool Existence
Utilize the `delegation_pool_exists` function to verify if a delegation pool is present at a given address.

2. Tracking Deposits
Once you confirm an account's deposit into a delegation pool, maintain a record of this deposit.

3. Calculating Rewards
With a record of deposits in place, leverage the `get_stake` function to ascertain the account's rewards.
